rotkehlchen/exchanges/kraken.py

```diff
@@ -638,50 +673,25 @@ def query_online_trade_history(
             start_ts: Timestamp,
             end_ts: Timestamp,
     ) -> Tuple[List[Trade], Tuple[Timestamp, Timestamp]]:
-        result, with_errors = self.query_until_finished(
-            endpoint='TradesHistory',
-            keyname='trades',
-            start_ts=start_ts,
-            end_ts=end_ts,
+        """
+        Query kraken events from database and create trades from them. May raise:
+        - RemoteError if the kraken pairs couldn't be queried
+        """
+        self.query_kraken_ledger_actions(start_ts=start_ts, end_ts=end_ts)
+        filter_query = HistoryEventFilterQuery.make(
+            from_ts=start_ts,
+            to_ts=end_ts,
+            event_type=[
+                HistoryEventType.TRADE,
+                HistoryEventType.RECEIVE,
+                HistoryEventType.SPEND,
+            ],
+            location=Location.KRAKEN,
+            location_label=self.name,
         )
-
-        # And now turn it from kraken trade to our own trade format
-        trades = []
-        max_ts = 0
-        for raw_data in result:
-            try:
-                new_trade = trade_from_kraken(raw_data)
-                if new_trade:
-                    trades.append(new_trade)
-                    max_ts = max(new_trade.timestamp, max_ts)
-            except UnknownAsset as e:
-                self.msg_aggregator.add_warning(
-                    f'Found kraken trade with unknown asset '
-                    f'{e.asset_name}. Ignoring it.',
-                )
-                continue
-            except UnprocessableTradePair as e:
-                self.msg_aggregator.add_error(
-                    f'Found kraken trade with unprocessable pair '
-                    f'{e.pair}. Ignoring it.',
-                )
-                continue
-            except (DeserializationError, KeyError) as e:
-                msg = str(e)
-                if isinstance(e, KeyError):
-                    msg = f'Missing key entry for {msg}.'
-                self.msg_aggregator.add_error(
-                    'Error processing a kraken trade. Check logs '
-                    'for details. Ignoring it.',
-                )
-                log.error(
-                    'Error processing a kraken trade',
-                    trade=raw_data,
-                    error=msg,
-                )
-                continue
-
-        queried_range = (start_ts, Timestamp(max_ts)) if with_errors else (start_ts, end_ts)
+        trades_raw = self.db.get_history_events(filter_query)
+        trades = self.process_kraken_trades(trades_raw)
+        queried_range = (start_ts, end_ts)
```

Comment:
So looking at this again you are removing functionality with this approach. Previously if for whatever reason we could not query all the range the query_until_done function returned the actual queried range, and this is what was written in the DB.

Yuo have to do the same here too. Essentially `query_kraken_ledger_actions` should return the actual range queried. For example if only half the range is queried due to rate limiting.

You can see the functionality in the diff as it was before. 

```python
queried_range = (start_ts, Timestamp(max_ts)) if with_errors else (start_ts, end_ts)
```

This `with_errors` from `query_until_finished` is what helps.